### PR TITLE
[WIP] Add Swagger Docs

### PR DESCRIPTION
**Pull request by DavisRayM**
### Changes / Features implemented

- Add Swagger docs

### TODO
- Silence warnings [warnings.txt](https://github.com/onaio/onadata/files/4657485/warnings.txt)
- Clean up code
- Configure OAuth in drf-spectacular settings https://drf-spectacular.readthedocs.io/en/latest/settings.html

### CODE REVIEW

File: onadata/apps/api/viewsets/data_viewset.py [hunk 1]
```diff
@@ -19,6 +19,8 @@
 from rest_framework.viewsets import ModelViewSet
 from rest_framework.exceptions import ParseError
 from rest_framework.settings import api_settings
+from drf_spectacular.utils import extend_schema, OpenApiParameter
+from drf_spectacular.types import OpenApiTypes
```
Comment (pld) [inline onadata/apps/api/viewsets/data_viewset.py:23]: Codacy Issue found: [Unused OpenApiTypes imported from drf_spectacular.types](https://app.codacy.com/manual/Ona/onadata/pullRequest?prid=5558448)

File: onadata/libs/serializers/dataview_serializer.py [hunk 2]
```diff
@@ -199,7 +200,7 @@ def get_count(self, obj):
 
         return None
 
-    def get_last_submission_time(self, obj):
+    def get_last_submission_time(self, obj) -> Optional[str]:
```
Comment (pld) [inline onadata/libs/serializers/dataview_serializer.py:203]: Codacy Issue found: [Method could be a function](https://app.codacy.com/manual/Ona/onadata/pullRequest?prid=5558448)

File: onadata/libs/serializers/dataview_serializer.py [hunk 3]
```diff
@@ -240,7 +241,7 @@ def get_instances_with_geopoints(self, obj):
 
         return False
 
-    def get_has_hxl_support(self, obj):
+    def get_has_hxl_support(self, obj) -> bool:
```
Comment (pld) [inline onadata/libs/serializers/dataview_serializer.py:244]: Codacy Issue found: [Method could be a function](https://app.codacy.com/manual/Ona/onadata/pullRequest?prid=5558448)

File: onadata/libs/serializers/dataview_serializer.py [hunk 4]
```diff
@@ -228,7 +229,7 @@ def get_last_submission_time(self, obj):
 
         return None
 
-    def get_instances_with_geopoints(self, obj):
+    def get_instances_with_geopoints(self, obj) -> bool:
```
Comment (pld) [inline onadata/libs/serializers/dataview_serializer.py:232]: Codacy Issue found: [Method could be a function](https://app.codacy.com/manual/Ona/onadata/pullRequest?prid=5558448)

File: onadata/apps/api/viewsets/data_viewset.py [hunk 5]
```diff
@@ -19,6 +19,8 @@
 from rest_framework.viewsets import ModelViewSet
 from rest_framework.exceptions import ParseError
 from rest_framework.settings import api_settings
+from drf_spectacular.utils import extend_schema, OpenApiParameter
```
Comment (pld) [inline onadata/apps/api/viewsets/data_viewset.py:22]: Codacy Issue found: [Unused extend_schema imported from drf_spectacular.utils](https://app.codacy.com/manual/Ona/onadata/pullRequest?prid=5558448)

File: onadata/libs/serializers/dataview_serializer.py [hunk 6]
```diff
@@ -174,7 +175,7 @@ def validate(self, attrs):
 
         return super(DataViewSerializer, self).validate(attrs)
 
-    def get_count(self, obj):
+    def get_count(self, obj) -> Optional[int]:
```
Comment (pld) [inline onadata/libs/serializers/dataview_serializer.py:178]: Codacy Issue found: [Method could be a function](https://app.codacy.com/manual/Ona/onadata/pullRequest?prid=5558448)

File: onadata/apps/api/viewsets/open_data_viewset.py [hunk 7]
```diff
@@ -171,7 +171,7 @@ def destroy(self, request, *args, **kwargs):
         return Response(status=status.HTTP_204_NO_CONTENT)
 
     @action(methods=['GET'], detail=True)
-    def schema(self, request, **kwargs):
+    def open_data_schema(self, request, **kwargs):
```
Comment (DavisRayM) [inline onadata/apps/api/viewsets/open_data_viewset.py:174]: The `drf-spectacular` package utilizes the `ViewSet.schema` attribute to retrieve schema information and this endpoint overrides that... We'll need to find another name for this endpoint to bypass the error with `drf-spectacular`

**Pull request closed without merging**